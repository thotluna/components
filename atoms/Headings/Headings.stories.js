import { Headings, options, styles } from './index'

import {
  getListTemplate,
  getTemplate,
  getOptionsArgTypes,
} from '../../helpers/storybook'

const Template = getTemplate(Headings, styles)
const ListTemplate = getListTemplate(Headings, styles)

export default {
  title: 'Atoms/Headings',
  component: Headings,
  args: {
    children: 'Consent doubt Tuckborough challenge destroying.',
  },
  argTypes: {
    color: getOptionsArgTypes(options.colors),
    size: getOptionsArgTypes(options.sizes),
    weight: getOptionsArgTypes(options.weights),
    children: { control: 'text' },
  },
}

export const Default = Template.bind({})

export const Colors = ListTemplate.bind({})
Colors.args = { items: options.colors.map((color) => ({ color })) }

export const Sizes = ListTemplate.bind({})
Sizes.args = { items: options.sizes.map((size) => ({ size })) }

export const Weights = ListTemplate.bind({})
Weights.args = { items: options.weights.map((weight) => ({ weight })) }
