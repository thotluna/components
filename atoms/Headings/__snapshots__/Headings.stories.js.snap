// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Atoms/Headings Colors 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <header
    className="headings color-base size-md weight-bold"
  >
    testing
  </header>
  <header
    className="headings color-primary size-md weight-bold"
  >
    testing
  </header>
  <header
    className="headings color-tertiary size-md weight-bold"
  >
    testing
  </header>
  <header
    className="headings color-inverted size-md weight-bold"
  >
    testing
  </header>
</div>
`;

exports[`Storyshots Atoms/Headings Default 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <header
    className="headings color-base size-md weight-bold"
  >
    testing
  </header>
</div>
`;

exports[`Storyshots Atoms/Headings Sizes 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <header
    className="headings color-base size-xs weight-bold"
  >
    testing
  </header>
  <header
    className="headings color-base size-sm weight-bold"
  >
    testing
  </header>
  <header
    className="headings color-base size-md weight-bold"
  >
    testing
  </header>
  <header
    className="headings color-base size-lg weight-bold"
  >
    testing
  </header>
  <header
    className="headings color-base size-xl weight-bold"
  >
    testing
  </header>
  <header
    className="headings color-base size-2xl weight-bold"
  >
    testing
  </header>
</div>
`;

exports[`Storyshots Atoms/Headings Weights 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <header
    className="headings color-base size-md weight-normal"
  >
    testing
  </header>
  <header
    className="headings color-base size-md weight-bold"
  >
    testing
  </header>
</div>
`;
